File: code/ch19/19.1.3. 189.rotate.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        t = None
        offset = n - k % n  # 右移变左移
        if offset == 0:
            return
        while offset:
            t = nums[0]
            offset -= 1
            for i in range(n - 1):
                nums[i] = nums[i + 1]
            nums[n - 1] = t

File: code/ch19/19.1.3. 189.rotate.cpp
#include <vector>
using namespace std;

class Solution {
public:
  void rotate(vector<int>& nums, int k) {
    int n = nums.size(), t, offset = n - k % n;
    if (offset == 0) return;
    while (offset) {
      t = nums[0];
      offset -= 1;
      for (int i = 0; i < n - 1; i++)
        nums[i] = nums[i + 1];
      nums[n - 1] = t;
    }
  }
};
